### PR TITLE
Add option for overwriting existing files (#64)

### PR DESCRIPTION
**Pull request by kalxas**
Implements #64

### CODE REVIEW

**Comment by pjonsson** [comment]
You should get an official opinion from someone who represents the project, but I feel the skip behavior should be the default (who wants to download a file they already have unless it's broken?), and the flag should be about forcing a download when a file already exists.

**Comment by kalxas** [comment]
Thank you @pjonsson for the feedback, I agree about the default behavior but I did not want to propose a change of it on my first pull request :)

**Comment by subnet001** [comment]
@kalxas Thanks for the contribuiton! I agree with you and @pjonsson about the default behavior, it seems like the right way of doing things. If you're up for it, please feel free to update your merge request to implement it, otherwise i can merge this as is for now, once it passes the pylint / black checks.

**Review by pedep — CHANGES_REQUESTED** [review]
Just a few changes to make the option a bit more accessible, otherwise LGTM :+1: 

I would also like to see the `os.path.join(path, filename.replace(".SAFE", ".zip"))` statement `DRY`ed, since its used twice now.

 something like
 
 ```python
 result_path = os.path.join(path, filename.replace(".SAFE", ".zip"))
 # [...]
 if skip and os.path.exists(result_path)
 # [...]
 shutil.move(tmp, result_path)
 ```

**Comment by kalxas** [comment]
Thank you, I will make the fixes in this PR and let you know to review

File: src/cdsetool/cli.py [hunk 1]
```diff
@@ -97,7 +98,7 @@ def download(
 
     list(
         download_features(
-            features, path, {"monitor": StatusMonitor(), "concurrency": concurrency}
+            features, path, {"monitor": StatusMonitor(), "concurrency": concurrency, "skip": skip}
```
[the remark this answers is not in the document]
Comment (kalxas) [reply src/cdsetool/cli.py]: Fixed

File: src/cdsetool/download.py [hunk 2]
```diff
@@ -27,8 +27,10 @@ def download_feature(feature, path, options=None):
     if not url or not filename:
         return feature.get("id")
 
-    # if os.path.exists(file):
-    #     return feature.get("id")
+    skip = options.get("skip",False)
+
+    if skip and os.path.exists(os.path.join(path, filename.replace(".SAFE", ".zip"))):
+        return feature.get("id")
```
[the remark this answers is not in the document]
Comment (kalxas) [reply src/cdsetool/download.py]: Fixed

**Comment by kalxas** [comment]
@subnet001 default is now to skip existing files.

**Comment by kalxas** [comment]
@subnet001 @pedep can you please enable the checks?

**Comment by kalxas** [comment]
Once this passes CI, I would be happy to squash the PR with a better commit message

**Review by pedep — APPROVED** [review]
Looks good to me :+1:

Thank you for your contribution :tada:

**Pull request merged**